### PR TITLE
Refactoring to less components and some bugfixes

### PR DESCRIPTION
**Pull request by gaabora**
Hello.
I did have to refactor code a bit to prevent some bugs when using SidebarMenu as standalone component

Cause I got bunch of _failed to mount component: template or render function not defined_ when
`import SidebarMenu from 'vue-sidebar-menu/components/SidebarMenu'`
to my wrapper component.

### CODE REVIEW

File: src/components/SidebarMenu.vue [hunk 1]
```diff
@@ -62,18 +62,20 @@
...
-            :show-child="showChild"
-            :rtl="rtl"
-            :is-collapsed="isCollapsed"
-          >
-            <slot
-              slot="dropdown-icon"
-              name="dropdown-icon"
-            />
-          </listItem>
+          <div class="vsm--list">
```
Comment (yaminncco) [inline src/components/SidebarMenu.vue]: v-if needed to be in the vsm--list
```suggestion
          <div class="vsm--list" v-if="mobileItem && mobileItem.child">
```

File: src/components/SidebarMenu.vue [hunk 2]
```diff
@@ -62,18 +62,20 @@
...
+            <sidebar-menu-item
```
Comment (yaminncco) [inline src/components/SidebarMenu.vue]: loop through `mobileItem.child` and level 2 prop
```suggestion
            <sidebar-menu-item
              v-for="(item, index) in mobileItem.child"
              :key="index"
              :level="2"
```

File: src/components/SidebarMenu.vue [hunk 3]
```diff
@@ -62,18 +62,20 @@
...
+              v-if="mobileItem && mobileItem.child"
+              :items="mobileItem.child"
```
Comment (yaminncco) [inline src/components/SidebarMenu.vue]:
```suggestion
              :item="item"
```

File: src/components/SidebarMenuItem.vue [hunk 4]
```diff
@@ -122,8 +122,10 @@
             v-if="show"
             class="vsm--dropdown"
           >
```
Comment (yaminncco) [inline src/components/SidebarMenuItem.vue]: `vsm--list` wrapper div missing

**Comment by gaabora** [comment]
Also I add a `defaultHref `property to let user override unwanted '#' in `itemLinkHref `(it is still '#' by default)

**Comment by gaabora** [comment]
> theres is no purpose to change '#' its a fallback when item href is not provided, so remove defaultHref and revert changes

I agree, and there no need in `a` if it's `href` is empty so I made one more refactoring to use `span` in that cases which will still act as a dropdown switch

**Pull request merged**